content/rs/references/client_references/client_csharp.md

```diff
@@ -1,61 +1,210 @@
 
 ## Installing StackExchange.Redis
 
-StackExchange.Redis' installation instructions are given in the ["Installation" section](https://stackexchange.github.io/StackExchange.Redis/) of its documentation page. It can be installed via the nuget package manager console with the following command:
+There are several ways to install this package including:
 
-    PM> Install-Package StackExchange.Redis
+1. With the .NET CLI: 
+```sh
+dotnet add package StackExchange.Redis
+```
+2. with the package manager console: 
+```sh
+`PM> Install-Package StackExchange.Redis`
+```
+3. With the NuGet GUI in Visual Studio   
 
 ## Opening a Connection to Redis Using StackExchange.Redis
 
-The following code creates a connection to Redis using StackExchange.Redis:
+The following code creates a connection to Redis using StackExchange.Redis in the context of a trivial console application:
 
-    using StackExchange.Redis;
+```csharp
+using StackExchange.Redis;
+using System;
+using System.Threading.Tasks;
+
+namespace ReferenceConsoleRedisApp
+{
+    class Program
+    {
+        static readonly ConnectionMultiplexer redis = ConnectionMultiplexer.Connect(
+            new ConfigurationOptions{
+                EndPoints = {"localhost:6379"}                
+            });
+        static async Task Main(string[] args)
+        {
+            var db = redis.GetDatabase();
+            var pong = await db.PingAsync();
+            Console.WriteLine(pong);
+        }
+    }
+}
+
+```
+
+The above example assumes that you have a Redis Server running locally.
+
+To configure the connection to your environment, adjust the parameters in the [ConfigurationOptions](https://stackexchange.github.io/StackExchange.Redis/Configuration) object appropriately. For the remainder of the examples the configuration will simply be `localhost`.
+
+## Connection Pooling with StackExchange.Redis
+
+While StackExchange.Redis does not provide direct means for conventional connection pooling, we recommend you **share and reuse** the ConnectionMultiplexer object. The ConnectionMultiplexer object should not be created per operation - it is to be created only once at the beginning and reused for the duration of the run. ConnectionMultiplexer is thread-safe so it can be safely shared between threads. For more information, refer to StackExchange.Redis’s [Basic Usage document](https://stackexchange.github.io/StackExchange.Redis/Basics).
+
+## Dependency Injection of the ConnectionMultiplexer
+
+As the `ConnectionMultiplexer` must be shared and reused within a runtime, it's recommended that you use [dependency injection](https://docs.microsoft.com/en-us/dotnet/core/extensions/dependency-injection) to pass it where it's needed. There's a few flavors of dependency injection depending on what you're using.
```

Comment:
When referring to specific classes, methods, etc, try to link to a doc explaining it on first use  Example: `[ConnectionManager](https://stackexchange.github.io/StackExchange.Redis/Basics)`.  This should be as close to a reference page as possible.  (Here, we have to link to the basics page, as there doesn't appear to be a reference page for the ConnectionMultiplexer class.)